File: go/example/telemetry.go
package main

import (
	"fmt"

	"github.com/betti-labs/betti-rdl"
)

func main() {
	kernel := bettirdl.NewKernel()
	defer kernel.Close()

	kernel.SpawnProcess(0, 0, 0)
	kernel.InjectEvent(0, 0, 0, 1)
	kernel.Run(100)

	tel := kernel.GetTelemetry()
	fmt.Printf("TELEMETRY,%d,%d,%d,%d\n", tel.EventsProcessed, tel.CurrentTime, tel.ProcessCount, tel.MemoryUsed)
}

File: rust/examples/telemetry.rs
use betti_rdl::Kernel;

fn main() {
    let mut kernel = Kernel::new();
    kernel.spawn_process(0, 0, 0);
    kernel.inject_event(0, 0, 0, 1);
    kernel.run(100);

    let tel = kernel.get_telemetry();
    println!(
        "TELEMETRY,{},{},{},{}",
        tel.events_processed, tel.current_time, tel.process_count, tel.memory_used
    );
}
